### PR TITLE
Turn priceFeed proposal into a SoftwareUpgrade

### LINKED ISSUE #9370: create proposal for upgrade16 to replace all priceFeed vats
## What is the Problem Being Solved?

#8400 reported that the priceFeed vats hold onto old quotes in their recoverySet, preventing them from being collected. #9013 made recoverySets optional. #8740 showed how to update the priceFeeds and their clients. #9283 applied the fixes to master. We have recent results showing that after these fixes, priceFeed vats and Zoe no longer grow as price updates are submitted.

image

## Description of the Design

The next step is to  apply the fixes from #9283 to an upcoming upgrade. (currently planned to be upgrade 16) Mainnet has different price feed vats, so the proposal will be different.

## Security Considerations

N/A

## Scaling Considerations

This  is all about remediating a known problem on the running chain.

## Test Plan

The approach has already been tested. We'll need to verify the script in a testNet, which will have yet other different priceFeeds.

## Upgrade Considerations

Yes

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #9370

## Description

The priceFeed proposal has been tested as an independent coreEval, but the current plan calls for it to be included in Upgrade 18. This adds it to `upgrade.go` and moves the tests to `n:upgrade-next`.

### Security Considerations

No additional security implications.

### Scaling Considerations

Addresses pre-existing scaling concerns (see #8400), and adds no new scaling issues.

### Documentation Considerations

None

### Testing Considerations

An existing test is moved to a new location.

The current version seems to tickle #10292, so tests may not pass at this point.

### Upgrade Considerations

Targeted for upgrade 18.

### CODE REVIEW

File: packages/builders/scripts/inter-protocol/updatePriceFeeds.js [hunk 1]
```diff
@@ -38,14 +38,21 @@ const configurations = {
...
 };
 
+const { keys } = Object;
+const Usage = `agoric run updatePriceFeed.js ${keys(configurations).join(' | ')}`;
 /** @type {import('@agoric/deploy-script-support/src/externalTypes.js').CoreEvalBuilder} */
 export const defaultProposalBuilder = async ({ publishRef, install }, opts) => {
+  const config = configurations[opts.variant];
+  if (!config) {
+    console.error(Usage);
+    process.exit(1);
```
Comment (dckc) [inline packages/builders/scripts/inter-protocol/updatePriceFeeds.js]: suggestion:
having `process.exit()` someplace other than the main function makes me uncomfortable. I'd rather see `throw` here.

**Review by dckc — APPROVED** [review]
`priceFeedUpdate.test.js` etc. seems like adequate test coverage. I did manage to summon attention to study the code in as much detail as I'd like, but I recognize a lot of the patterns, and if the tests pass, I'm reasonably confident that it's good.

I much prefer that #10317 is merged first. I expect @frazarshad can get that done in the next day.

Merging this into fraz/replace-committee-a3p will complicate review of #10317; I'd rather we don't do that, but I find it acceptable.

File: packages/builders/scripts/inter-protocol/updatePriceFeeds.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/builders/scripts/inter-protocol/updatePriceFeeds.js]: fixed.

**Comment by Chris-Hibbert** [comment]
cc: @frazarshad 
 Since the auctioneer (instance and governor) are now being updated in the same software update, the EC update has to make use of those new values.  Please take a look at how I connected them.

**Comment by frazarshad** [comment]
> Since the auctioneer (instance and governor) are now being updated in the same software update, the EC update has to make use of those new values. Please take a look at how I connected them.

LGTM 👍

**Pull request merged**